syntax = "proto3";

package common;

service LockdownService {
  // Detection calls this on Backup and Clients
  rpc TriggerLockdown (LockdownRequest) returns (LockdownResponse) {}
}

message LockdownRequest {
  string threat_id = 1;
  string timestamp = 2;
  string reason = 3;
  string targeted_node = 4;
}

message LockdownResponse {
  bool success = 1;
  string status_message = 2;
}